cpp/tests/reductions/reduction_tests.cpp

```diff
@@ -1081,6 +1081,26 @@ TYPED_TEST(FixedPointTestBothReps, FixedPointReductionProduct)
   }
 }
 
+TYPED_TEST(FixedPointTestBothReps, FixedPointReductionProductWithNulls)
+{
+  using namespace numeric;
+  using decimalXX  = TypeParam;
+  using RepType    = cudf::device_storage_type_t<decimalXX>;
+  using fp_wrapper = cudf::test::fixed_point_column_wrapper<RepType>;
+
+  for (int i = -1; i <= 0; ++i) {
+    auto const scale    = scale_type{i};
```

Comment:
How about `for (auto const& s: {-1,0})` or `for (auto const& scale: {scale_type{-1}, scale_type{0}})` ?